support/cas-server-support-duo-core/src/main/java/org/apereo/cas/adaptors/duo/authn/DuoAuthenticationMetaDataPopulator.java

```diff
@@ -0,0 +1,38 @@
+package org.apereo.cas.adaptors.duo.authn;
+
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.apereo.cas.authentication.AuthenticationBuilder;
+import org.apereo.cas.authentication.AuthenticationManager;
+import org.apereo.cas.authentication.AuthenticationTransaction;
+import org.apereo.cas.authentication.Credential;
+import org.apereo.cas.authentication.metadata.BaseAuthenticationMetaDataPopulator;
+
+/**
+ * Implementation of AuthenticationMetadaPopulator to handle multiple Duo instances.
+ *
+ * @author Travis Schmidt
+ * @since 5.3.4
+ */
+@Slf4j
+@ToString(callSuper = true)
+@RequiredArgsConstructor
+public class DuoAuthenticationMetaDataPopulator extends BaseAuthenticationMetaDataPopulator {
+    private final String authenticationContextAttribute;
+
+    private final DuoAuthenticationHandler authenticationHandler;
```

Comment:
Could this be refactored now to use a more generic type? Since you're only adding a name here? And is this not a duplicate of `AuthenticationContextAttributeMetaDataPopulator`?

Reply:
Yes, it has been refactored into a duplicate of AuthenticationContextAttributeMetaDataPopulator as I worked through the changes.  I will remove it.